shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,1079 @@
+
+    In case of product tracked by lot, you will have to enter the lot number and its
+    expiry date (unless it is already known by the system).
+
+    Moves are not validated as they are processed. It is the responsibility of the
+    user to decide when to mark as done already processed lines.
+    Any remaining lines will be pushed to a backorder.
+    """
+
+    _inherit = "base.shopfloor.process"
+    _name = "shopfloor.reception"
+    _usage = "reception"
+    _description = __doc__
+
+    def _move_line_by_product(self, product):
+        return self.env["stock.move.line"].search(
+            self._domain_move_line_by_product(product)
+        )
+
+    def _move_line_by_packaging(self, packaging):
+        return self.env["stock.move.line"].search(
+            self._domain_move_line_by_packaging(packaging)
+        )
+
+    # DOMAIN METHODS
+
+    def _domain_move_line_by_packaging(self, packaging):
+        return [
+            ("move_id.picking_id.picking_type_id", "in", self.picking_types.ids),
+            ("move_id.picking_id.state", "=", "assigned"),
+            ("move_id.picking_id.user_id", "=", False),
+            ("package_id.product_packaging_id", "=", packaging.id),
+        ]
+
+    def _domain_move_line_by_product(self, product):
+        return [
+            ("move_id.picking_id.picking_type_id", "in", self.picking_types.ids),
+            ("move_id.picking_id.state", "=", "assigned"),
+            ("move_id.picking_id.user_id", "=", False),
+            ("product_id", "=", product.id),
+        ]
+
+    def _domain_stock_picking(self):
+        return [
+            ("state", "=", "assigned"),
+            ("picking_type_id", "in", self.picking_types.ids),
+            ("user_id", "=", False),
+        ]
+
+    def _select_picking(self, picking):
+        if picking.picking_type_id not in self.picking_types:
+            return self._response_for_select_document(
+                message=self.msg_store.cannot_move_something_in_picking_type()
+            )
+        if picking.state != "assigned":
+            return self._response_for_select_document(
+                message=self.msg_store.stock_picking_not_available(picking)
+            )
+        return self._response_for_select_line(picking)
+
+    def _response_for_select_line(self, picking, message=None):
+        picking.user_id = self.env.user
+        data = {"picking": self._data_for_stock_picking(picking, with_lines=True)}
+        return self._response(next_state="select_line", data=data, message=message)
+
+    def _response_for_confirm_done(self, picking, message=None):
+        picking.user_id = self.env.user
+        data = {"picking": self._data_for_stock_picking(picking, with_lines=True)}
+        return self._response(next_state="confirm_done", data=data, message=message)
+
+    def _response_for_confirm_new_package(
+        self, picking, line, new_package_name, message=None
+    ):
+        picking.user_id = self.env.user
```

Comment:
I would not set the user here. This should be one explicitly by the endpoint, not by the method collecting data for the response.